extensions/wikia/EnableDiscussions/EnableDiscussionsHooks.class.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+
+class EnableDiscussionsHooks {
+
+	const DISCUSSIONS_URL = '/d/f';
+	// Forum namespaces should be available
+	// because both Forum.setup and ForumDisabled.setup define them
+	private static $forumNamespaces = [
+		NS_WIKIA_FORUM_BOARD,
+		NS_WIKIA_FORUM_TOPIC_BOARD,
+		NS_WIKIA_FORUM_BOARD_THREAD,
+	];
+
+
+	public static function onBeforeInitialize(
+		Title $title = null, $unused, OutputPage $output = null, $user, $request, $mediaWiki
+	) {
+		if ( !self::shouldRedirectToDiscussions() || $title == null ) {
+			return true;
+		}
+		$forumTitle = self::getForumTitle( $title );
+		if ( empty( $forumTitle ) ) {
+			return true;
+		}
+
+		$output->redirect( self::getDiscussionsUrl( $forumTitle->getNamespace(),
+			$forumTitle->getArticleID() ) );
+
+		return false;
+	}
+
+	/**
+	 * @return bool
+	 */
+	private static function shouldRedirectToDiscussions() {
+		global $wgEnableForumExt, $wgEnableDiscussions;
+
+		return empty( $wgEnableForumExt ) && $wgEnableDiscussions;
+	}
+
+	/**
+	 * @param $title
+	 * @return null|Title
+	 */
+	public static function getForumTitle( Title $title ) {
+		if ( $title->isSpecial( 'Forum' ) ||
+		     in_array( $title->getNamespace(), self::$forumNamespaces )
+		) {
+			return $title;
+		}
+		if ( $title->getNamespace() === NS_USER_WALL_MESSAGE ) {
+			$mainTitle = Title::newFromID( $title->getText() );
+			/*
+			 * if we visit the thread using Thread:xxxxx url, $wgTitle's namespace is 1201 for both, wall and forum
+			 * threads. But when we create new title from it using newFromId, we're able to distinguish those two
+			 */
+			if ( $mainTitle instanceof Title &&
+			     in_array( $mainTitle->getNamespace(), self::$forumNamespaces )
+			) {
+				return $mainTitle;
+			}
+		}
+		if ( $title->getNamespace() === 0 ) {
```

Comment:
One FYI, namespace zero is the main namespace and can be compared to `NS_MAIN`.  I'm assuming this check is needed when the Forum extension is disabled and the `Board` namespace no longer registers as a defined namespace (since `Forum.namespaces.php` isn't loaded)?  Maybe we could refactor this and the above code to use the same function since they're almost identical.

As a history lesson, in the above code `$title->getText()` is getting the article ID.  The `getText()` call gets the "text" part of the title which does not include the namespace.  So for example, for "/wiki/Thread:12345" the text part would be "12345".  This is the title's article ID, as the folks who created Forums embedded there, a convention for Forms.  This is non-standard for MW and weird, but that's why `getText` for this type of title will get you the article ID, and avoids an extra DB lookup that would be necessary.

For the case on the above line (L64) where namespace is `NS_MAIN`, its likely a path like '/wiki/Board:Fun_And_Games' where `Board` is no longer defined as a namespace.  This requires a `getArticleID()` call (which does a DB query) in order to get this article ID.